core/types/transaction.go

```diff
@@ -488,6 +503,47 @@ func (t *TransactionsByPriceAndNonce) CurrentSize() int {
 	return len(t.heads)
 }
 
+//Forward move t to be one index behind tx, tx cant be nil
+func (t *TransactionsByPriceAndNonce) Forward(tx *Transaction) {
+	if tx == nil {
+		txTmp := t.Peek()
+		for txTmp != nil {
+			t.Shift()
+			txTmp = t.Peek()
+		}
+		return
+	}
+
+	l := len(t.heads)
+	acc, _ := Sender(t.signer, tx)
+	for i := 0; i < l; i++ {
+		accTmp, _ := Sender(t.signer, t.heads[i])
+		if acc == accTmp {
+			if tx == t.heads[i] {
+				txTmp := t.Peek()
```

Comment:
If prefethcer is faster than miner worker, the logic here will drain all txs.

Reply:
Is that not the expected behavior as we discussed ? :
only one of two conditions will stop the `prefetcher` routine:
- all txs been executed
- `worker` routine finished its job